std/stdio.d

```diff
@@ -4522,70 +4521,70 @@ Initialize with a message and an error code.
     }
 }
 
-extern(C) void std_stdio_static_this()
+// Undocummented but public because std* handles are aliasing it
+ref File makeGlobal(alias handle)()
 {
-    static import core.stdc.stdio;
-    //Bind stdin, stdout, stderr
-    __gshared File.Impl stdinImpl;
-    stdinImpl.handle = core.stdc.stdio.stdin;
-    .stdin._p = &stdinImpl;
-    // stdout
-    __gshared File.Impl stdoutImpl;
-    stdoutImpl.handle = core.stdc.stdio.stdout;
-    .stdout._p = &stdoutImpl;
-    // stderr
-    __gshared File.Impl stderrImpl;
-    stderrImpl.handle = core.stdc.stdio.stderr;
-    .stderr._p = &stderrImpl;
+    static __gshared File.Impl impl;
+    static __gshared File result;
+    static shared bool initialized;
+    if (!initialized)
+    {
+        // Use double-checking because initOnce is inefficient.
```

Comment:
initOnce can't optimize it. It has to take the (potentially) fenced approach in order to make sure it does indeed store the initializer only once, and that may be slower on non-x86. It may be **called** multiple times, but only once should it actually store. But there is no reason to call it at all, and thus incur a penalty, if `true` is already observed.

@schveiguy, initializer is stored under a mutex, which should imply a fence. Sigh... "should". @klickverbot is right, we need a formalized model.